rust/index/src/spann/types.rs

```diff
@@ -1,40 +1,71 @@
-use std::collections::HashMap;
+use std::{
```

Comment:
Please add more comments here - its critical code we ought to make extremely readable to future readers. Connect to the papers terminology where possible. Connect to the other OSS impl where possible

Reply:
Added a task in tracker. Will do for all the code in one PR. It will also be a good opportunity to revisit the code for any obvious oversights/bugs